sway-error/src/error.rs

```diff
@@ -1138,7 +1138,11 @@ impl ToDiagnostic for CompileError {
                     format!("Consider removing the variable \"{variable}\" altogether, or adding it to all alternatives."),
                 ],
             },
-            TraitNotImportedAtFunctionApplication { trait_name, function_name, function_call_site_span, trait_constraint_span, trait_candidates }=> Diagnostic {
+            TraitNotImportedAtFunctionApplication { trait_name, function_name, function_call_site_span, trait_constraint_span, trait_candidates }=> {
+                // Make candidates order deterministic
+                let mut trait_candidates = trait_candidates.clone();
+                trait_candidates.sort();
```

Comment:
I am thinking of adding `let trait_candidates = &trait_candidates; // Remove immutability.` here, just to remove the mutability for the rest of the diagnostics and have clear semantics. Safety first, but yes I am sometimes a bit too paranoid :-)

Reply:
Good idea, added in #[497468b](https://github.com/FuelLabs/sway/pull/5400/commits/497468b545c9ba0d890915bddf0e4af15f575932)